--- vnstat-1.11/src/vnstat.c.old	2011-06-01 00:29:51.000000000 +0200
+++ vnstat-1.11/src/vnstat.c	2013-08-07 14:42:58.396231450 +0200
@@ -485,7 +485,7 @@
 			readdb(interface, dirname);
 
 			data.totalrx=data.totaltx=data.totalrxk=data.totaltxk=0;
-			for (i=0; i<=29; i++) {
+			for (i=0; i<=12; i++) {
 				if (data.month[i].used) {
 					addtraffic(&data.totalrx, &data.totalrxk, data.month[i].rx, data.month[i].rxk);
 					addtraffic(&data.totaltx, &data.totaltxk, data.month[i].tx, data.month[i].txk);
